ts/components/messages/Inbox.tsx

```diff
@@ -0,0 +1,225 @@
+import { fromNullable, none } from "fp-ts/lib/Option";
+import * as pot from "italia-ts-commons/lib/pot";
+import { Button, Text, View } from "native-base";
+import React from "react";
+import { FlatList, ListRenderItem, StyleSheet } from "react-native";
+
+import I18n from "../../i18n";
+import { lexicallyOrderedMessagesStateInfoSelector } from "../../store/reducers/entities/messages";
+import { MessageState } from "../../store/reducers/entities/messages/messagesById";
+import { paymentsByRptIdSelector } from "../../store/reducers/entities/payments";
+import { servicesByIdSelector } from "../../store/reducers/entities/services/servicesById";
+import InboxItem from "./InboxItem";
+
+export const styles = StyleSheet.create({
+  listWrapper: {
+    flex: 1
+  },
+  buttonBar: {
+    position: "absolute",
+    left: 0,
+    right: 0,
+    bottom: 0,
+    flexDirection: "row",
+    zIndex: 1,
+    justifyContent: "space-around",
+    backgroundColor: "#ddd",
+    padding: 10,
+    opacity: 0.75
+  },
+
+  buttonBarButton: {
+    opacity: 1
+  }
+});
+
+type Props = {
+  messagesStateInfo: ReturnType<
+    typeof lexicallyOrderedMessagesStateInfoSelector
+  >;
+  servicesById: ReturnType<typeof servicesByIdSelector>;
+  paymentsByRptId: ReturnType<typeof paymentsByRptIdSelector>;
+  refreshMessages: () => void;
+  setMessagesArchivedState: (
+    ids: ReadonlyArray<string>,
+    archived: boolean
+  ) => void;
+  onPressItem: (id: string) => void;
+};
+
+type State = {
+  lastMessageStatesUpdate: number;
+  filteredMessageStates: ReturnType<typeof generateFilteredMessageStates>;
+  isSelectionModeEnabled: boolean;
+  selected: Map<string, true>;
+};
+
+const inboxItemKeyExtractor = (messageState: MessageState) =>
+  messageState.meta.id;
+
+/**
+ * Filter only the messages that are not archived.
+ * @param potMessageStates
+ */
+const generateFilteredMessageStates = (
+  potMessageStates: pot.Pot<ReadonlyArray<MessageState>, string>
+): ReadonlyArray<MessageState> => {
+  if (pot.isSome(potMessageStates)) {
+    return potMessageStates.value.filter(
+      messageState => !messageState.isArchived
+    );
+  }
+
+  return [];
+};
+
+/**
+ * A component to render a list of not archived messages.
+ * The component allows messages selection and archiving.
+ */
+class Inbox extends React.PureComponent<Props, State> {
+  /**
+   * The function is used to update the filteredMessageStates only when necessary.
+   */
+  public static getDerivedStateFromProps(
```

Comment:
this use of `getDerivedStateFromProps ` looks like an antipattern, see: https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#when-to-use-derived-state

> If you’re using derived state to memoize some computation based only on the current props, you don’t need derived state.

Reply:
The detail that make it not the antipatern of the blogpost is that i am not only using nextProps but also prevState (lastMessageStatesUpdate) to apply the filter conditionally.